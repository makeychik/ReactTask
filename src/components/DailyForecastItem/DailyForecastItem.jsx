import React from 'react';
import WeatherSymbol from '../WeatherSymbol/WeatherSymbol';
import classes from './DailyForecastItem.module.scss';

const DailyForecastItem = ({ forecastItem }) => {
  const forecastTime = forecastItem.time.match(/\d\d:\d\d/);

  return (
    <div className={classes.dailyForecastItem}>
      <h2 className={classes.dailyForecastItem__header}>{forecastTime}</h2>
      <WeatherSymbol symbol={forecastItem.symbol ?? 'n400'} />
      <span>{forecastItem.temperature} Â°C</span>
    </div>
  );
};

export default DailyForecastItem;
