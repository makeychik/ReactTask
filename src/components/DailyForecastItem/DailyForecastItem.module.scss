.dailyForecastItem {
  border: 1px solid black;
  border-radius: 5px;
  padding: 10px;
  margin: 5px;
  width: 23%;
}

.dailyForecastItem__header {
}

.symbol {
  height: 100px;
}
