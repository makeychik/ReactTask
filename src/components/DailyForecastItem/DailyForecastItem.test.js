import React from 'react';
import { render } from '@testing-library/react';
import DailyForecastItem from './DailyForecastItem';

const forecastItem = {
  time: '2021-12-20T13:00+04:00',
  symbol: 'd400',
  temperature: -0.78
};

describe('DailyForecastItem component', () => {
  it('component rendered successfully and match the snapshot', () => {
    const component = render(<DailyForecastItem forecastItem={forecastItem} />);
    expect(component).toMatchSnapshot();
  });
});
