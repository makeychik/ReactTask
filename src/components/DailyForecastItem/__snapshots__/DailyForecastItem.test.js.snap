// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyForecastItem component component rendered successfully and match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dailyForecastItem"
      >
        <h2
          class="dailyForecastItem__header"
        >
          13:00
        </h2>
        <img
          class="symbol"
          src="https://developer.foreca.com/static/images/symbols/d400.png"
        />
        <span>
          -0.78
           °C
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dailyForecastItem"
    >
      <h2
        class="dailyForecastItem__header"
      >
        13:00
      </h2>
      <img
        class="symbol"
        src="https://developer.foreca.com/static/images/symbols/d400.png"
      />
      <span>
        -0.78
         °C
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
